### PR TITLE
Wrap some SysMemForKernel's nids, fixing #7960

### PR DESCRIPTION
**Pull request by sum2012**
#7960

The log is better
https://gist.github.com/sum2012/2ab67196dc4a97ff60e7

### CODE REVIEW

**Comment by sum2012** [comment]
@hrydgard this commit have problem ?

**Comment by sum2012** [comment]
OMG,If I include sceKernelMemory.cpp

**Comment by sum2012** [comment]
I move SysMemForKernel to sceKernelMemory 
As SysMemForUser in there

File: Core/HLE/sceKernelMemory.cpp [hunk 1]
```diff
@@ -2301,3 +2301,96 @@ void Register_SysMemUserForUser()
...
+	HeapInformation() {}
+	int uid;
+	int partitionId;
+	u32 size;
+	int flags;
+	u32 address;
+	std::string name;
+	BlockAllocator alloc;
+	static int GetStaticIDType() { return SCE_KERNEL_TMID_Fpl; }//  wrong
+	int GetIDType() const override { return SCE_KERNEL_TMID_Fpl; }//  wrong
```
Comment (sum2012) [inline Core/HLE/sceKernelMemory.cpp]: I take a search but no SCE_KERNEL_TMID_HEAP

**Comment by sum2012** [comment]
Unfortunately,the screen is still black after HLE\sceKernelThread.cpp:2851 sceKernelSuspendThread(276): cannot suspend current thread

modify log:
https://gist.github.com/sum2012/04b57eb492018d02ca29

**Comment by sum2012** [comment]
In the soft gpu,the menu appear now.
(But in-game crash even turn off fast memory)

File: Core/HLE/sceKernelMemory.cpp (hunk 1 above)
Comment (sum2012) [inline Core/HLE/sceKernelMemory.cpp]: partitionId, name,flags,name not used
Not sure really need save status these variable

**Comment by sum2012** [comment]
The crash is from sceKernelDeleteHeap...
I cannot lend code from sceheap as it doesn't use uid

**Comment by sum2012** [comment]
Just a note my change is based on https://github.com/jpcsp/jpcsp/commit/3c25717e04c66a1f5a93e0cacc2c232ecf2d2453

File: Core/HLE/sceKernelMemory.cpp (hunk 1 above)
Comment (unknownbrackets) [inline Core/HLE/sceKernelMemory.cpp]: We don't use the "XyzInformation" convention in PPSSPP, even if JPCSP does.  I really think we should have testing to understand how this differs from sceHeap.  Also, testing can identify which TMID is uses - there's a syscall that gives you the type id for a uid, iirc.

I can't remember for sure, but I thought we used the TMID for savestates, though.  I don't think it's good if they're wrong.

-[Unknown]

Comment (sum2012) [inline Core/HLE/sceKernelMemory.cpp]: So this pull request is not ready for v1.1

Comment (sum2012) [inline Core/HLE/sceKernelMemory.cpp]: Also yes,I copy most of code from sceHeap.cpp

Comment (hrydgard) [inline Core/HLE/sceKernelMemory.cpp]: yeah, it won't make it. we can start adding some kernel stuff after 1.1 but for now I want to keep things the way they are to be safe.

1.1 will be soon though, I'm planning to release next weekend.

File: Core/HLE/sceKernelMemory.cpp [hunk 2]
```diff
@@ -2301,3 +2301,115 @@ void Register_SysMemUserForUser()
 {
 	RegisterModule("SysMemUserForUser", ARRAY_SIZE(SysMemUserForUser), SysMemUserForUser);
 }
+
+
+struct HeapInformation : public KernelObject {
```
Comment (unknownbrackets) [inline Core/HLE/sceKernelMemory.cpp]: I think we should:

1. Put these functions and structs in `sceKernelHeap.cpp`.
2. Keep `SysMemForKernel` here and export the functions in `sceKernelHeap.h` (kinda like sceKernelMutex.h, etc.)

-[Unknown]

File: Core/HLE/sceKernelMemory.cpp (hunk 1 above)
Comment (unknownbrackets) [inline Core/HLE/sceKernelMemory.cpp]: See also `PPSSPP_KERNEL_TMID_Module` and similar for using fake TMID values.  For savestates to work correctly, we need unique values right now.

-[Unknown]

**Comment by hrydgard** [comment]
This will need some rebasing and cleanup work but seems worthwhile if it gets Chinese language patches going, with not too much trouble.

@sum2012  do you intend to improve this some more anytime soon? Otherwise I might clean it up a bit soon for you, but you'll have to test it.

**Comment by sum2012** [comment]
@hrydgard I remember this code not more work , maybe I transate something wrong
with https://github.com/jpcsp/jpcsp/commit/3c25717e04c66a1f5a93e0cacc2c232ecf2d2453

**Comment by hrydgard** [comment]
Hm. Either way it needs a rebase if we should keep it open. Close or not?

**Comment by hrydgard** [comment]
This needs a rebase, and Unknown made comments that need addressing. Do you plan to work on it anytime soon?

**Comment by sum2012** [comment]
I would rebase it in Sunday in Hong Kong time

**Comment by hrydgard** [comment]
Oh, there's absolutely no rush. Whenever you have time.

**Comment by sum2012** [comment]
FPL is from sceKernelMemory,I don't know how to solve
kernelObjects.Destroy<heap>(heap->uid); not work ether

**Comment by hrydgard** [comment]
Closing this in favor of #12225.

Sorry for the long delay ;)

**Pull request closed without merging**